cos/static/board_builder.js

```diff
@@ -130,11 +142,100 @@ function build_board() {
 
                 layer.add(settlement_area_right);
             }
+            settlement_area_right.on('mouseup', function(){
+                if(settlement_area_right.getFill() == 'red'){
+                    place_settlement(this.x(),this.y(), settlementX, settlementY, stage, layer);
+                }
+            })
+
+        }
+    }
 
+    for (var i = 0; i < 6; i++) {
```

Comment:
Could you add comments to the code to highlight what generally is happening here?